src/components/NFTViewOptionList.vue

```diff
@@ -183,50 +184,44 @@ export default {
           return this.$t('nft_details_page_button_view_unknown');
       }
     },
-    getFilenameFromURL(url) {
-      const qsStr = url.split('?').pop();
-      const qs = querystring.parse(qsStr);
-      return qs?.name || '';
-    },
-    getDownloadFilenameFromURL(url) {
-      return `${this.getFilenameFromURL(url) || 'content'}`;
-    },
+    getFilenameFromURL,
+    getDownloadFilenameFromURL,
     handleClickViewContent() {
       this.$emit('view-content');
     },
+    async passCORSPolicy(url) {
+      try {
+        const res = await this.$axios.get(url, { responseType: 'stream' });
+        if (res.status === 200) {
+          res.data.destroy();
+        }
+        return true;
+      } catch (error) {
+        return false;
+      }
+    },
     async handleClickViewContentURL(e, contentUrl) {
       const type = this.getContentUrlType(contentUrl);
       const url = parseNFTMetadataURL(contentUrl);
       this.$emit('view-content-url', e, url, type);
-      if (type === 'pdf') {
-        e.preventDefault();
-        this.$router.push(
-          this.localeLocation({
-            name: 'reader',
-            query: {
-              download: this.isContentDownloadable ? '1' : '0',
-              classId: this.classId,
-              format: type,
-              src: url,
-            },
-          })
-        );
-      } else if (type === 'epub') {
+      if (['pdf', 'epub'].includes(type)) {
         e.preventDefault();
-        // NOTE: Allow user to download epub file before we have web epub reader
-        try {
-          this.alertPromptSuccess(this.$t('nft_download_content_prepare'));
-          const blob = await this.$axios.$get(url, { responseType: 'blob' });
-          saveAs(blob, this.getDownloadFilenameFromURL(contentUrl));
-        } catch (error) {
-          if (error.message === 'Network Error') {
-            // bypass CORS
-            window.location.href = url;
-          } else {
-            // eslint-disable-next-line no-console
-            console.error(error);
-            this.alertPromptError(this.$t('nft_download_content_error'));
-          }
+        const passCORS = await this.passCORSPolicy(url);
+        if (passCORS) {
```

Comment:
On second thought, why check CORS here? We can always handle it in reader page for pdf and epub using the cors proxy cloud function